test/spec/controllers/courtSearchInfo.ctrl.spec.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+describe('CourtSearchInfoCtrl', function() {
+  var CourtSearchInfoCtrl;
+  var sample_court;
+
+  sample_court = {
+    id: 9,
+    latitutde: 38.7947506,
+    longitude: -90.26261696,
+    municipality: "Black Jack",
+    address: "12500 Old Jamestown Road",
+    city: "St. Louis",
+    state: "MO",
+    zip_code: 63033
+  };
+
+  beforeEach(function() {
+    module('yourStlCourts');
+
+    inject(function($controller,$state, $window, $q, $httpBackend){
+
+      CourtSearchInfoCtrl = $controller('CourtSearchInfoCtrl',{
+        $state: $state,
+        $window: $window,
+        court:sample_court
+      });
+
+      $httpBackend.whenGET(/courts/).respond(200, '');
+    });
+  });
+
+  xit('goes back to home page if court is null',inject(function($state){
+    spyOn($state,'go');
+    sample_court = null;
+    expect($state.go).toHaveBeenCalledWith('home');
+  }));
+
+  it('prints if printCourtInfo is called',inject(function($window){
+    spyOn($window,'print');
+    CourtSearchInfoCtrl.printCourtInfo();
+    expect($window.print).toHaveBeenCalled();
+  }));
+
+  it('sets court on initialization',inject(function($rootScope){
+    $rootScope.$apply();
+    expect(CourtSearchInfoCtrl.courtInfo).toEqual(sample_court);
+  }));
+
+  it('calls and gets correct directions',(function(){
```

Comment:
Oh I see. Well, if that function is "public" (aka on "ctrl"), then it also needs to be tested individually. In your case, you want to verify that it was invoked on startup - which belongs in a separate test (something like it('initializes with court info....'); In that case, you to spy on the method **before** it's invoked. There is a way in $controller to not immediately initialize the controller but it's not really documented. After your injections, you can pass ```true``` as a 3rd param to the $controller function":

`var ctrlFunc = $controller('MyCtrl', {
  injectable: myVar,
  ....
}, true);```

then you can spy or set variables on ```ctrlFunc.instance```.

i.e. ```spyOn(ctrlFunc.instance.getCourtDirectionLink)```

Then to actually init the controller as you would normally:
```MyCtrl = ctrlFunc();```

You'll want to do that in your beforeEach as a slight adjustment from what you have now.